pkg/middleware/middleware.go

```diff
@@ -0,0 +1,61 @@
+package middleware
```

Comment:
Not sure if this scope is too narrow, what do you think about a broader `http` package to include everything related to our customized http server?

Reply:
Your call, we can always change the name in the future.

For me it's uncomfortable to see the `http` server logic in `root.go` and I think that we should probably move that somewhere else (hence the broader scope for this package, to cover everything related to http/internalhttp).

In any case, this can be covered in the future.

Reply:
Renamed to http, because it is indeed better. We can move server logic under http in a followup PR.